pkg/resourcemanager/controller/garbagecollector/reconciler.go

```diff
@@ -113,6 +111,11 @@ func (r *Reconciler) Reconcile(reconcileCtx context.Context, _ reconcile.Request
 		objList := &metav1.PartialObjectMetadataList{}
 		objList.SetGroupVersionKind(gvk)
 		if err := r.TargetReader.List(ctx, objList); err != nil {
+			if meta.IsNoMatchError(err) && gvk == mrGVK {
+				// ignore the error if the resource is not matched since
+				// not all clusters have the managed resource CRD installed
```

Comment:
My reasoning was that we do not expect any of the other requests to return a `no match error` so if that happens we want to report it since it is unexpected behavior.